docs/source/migration.rst

```diff
@@ -1,7 +1,31 @@
 1.0 Migration Guide
 ===================
 
-The following guide illustrates the changes in applications written with ``0.x`` version of Beaker compared to ``1.0``
+The following guide illustrates the changes in applications written with ``0.x`` version of Beaker compared to ``1.0``.
+The rationale behind these changes is described by the `AlgoKit Beaker productionisation review architecture decision record <https://github.com/algorandfoundation/algokit-cli/blob/main/docs/architecture-decisions/2023-01-11_beaker_productionisation_review.md>`_.
+
+The goal of these changes is to create a Beaker experience that is easier to use, less surprising and resolves some key bugs that were identified.
+
+Recommended migration approach
+------------------------------
+
+Any migration effort requires careful consideration to ensure it doesn't result in breaking changes.
+We have carefully articulated all considerations in this document that should be evaluated to perform a
+successful migration of a ``0.x`` Beaker contract/signature to a ``1.0`` compatible one.
+
+While there are a number of considerations to work through, care was taken to ensure that the migration
+effort should be relatively straightforward. If you have any problems or questions about a migration feel
+free to send messages to the `algokit channel in Algorand Discord <https://discord.com/channels/491256308461207573/1065320801970180168>`_.
+
+In order to de-risk the migration effort we recommend you test for output stability of the TEAL code
+that is generated from your Beaker code. This involves storing the current TEAL and application spec output from calling
+``.dump()`` on your Beaker contract and then doing a diff after performing the migration. We expect there will be
```

Comment:
.export() now, see Application.compile() section

Reply:
Actually to get a diff, it is ``.dump()`` on the original ``0.x`` application and then ``.export()`` on the specification returned from ``.build()`` after migration to ``1.0``